src/logic/Applications.swift

```diff
@@ -104,7 +105,30 @@ class Applications {
     private static func isActualApplication(_ app: NSRunningApplication) -> Bool {
         // an app can start with .activationPolicy == .prohibited, then transition to != .prohibited later
         // an app can be both activationPolicy == .accessory and XPC (e.g. com.apple.dock.etci)
-        return (isNotXpc(app) || isAndroidEmulator(app)) && !app.processIdentifier.isZombie()
+        return (isNotXpc(app) || isAndroidEmulator(app)) && !app.processIdentifier.isZombie() && isAnWindowApplication(app)
```

Comment:
If your comment below is correct (that apps that trigger a `didActivateApplicationNotification` event must have windows), then all these checks are no longer useful for these apps, no?

I imagine that they would only be useful for the `wasLaunchedBeforeAltTab == true` apps. So maybe we should add that boolean check first (left-side), to avoid the more expansive checks in the case where it's `wasLaunchedBeforeAltTab == false`, no?